tests/robots/test_configuration.py

```diff
@@ -6,27 +7,43 @@
 from compas_fab.robots import to_degrees
 
 
+def test_bool():
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'c'])
+    assert config
+
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3)
+    assert config
+
+
+def test_len():
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'c'])
+    assert len(config) == 3
+
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3)
+    assert len(config) == 0
```

Comment:
I don't like this much. I understand why it happens, but it does not make much sense, especially if you consider that names are optional. Should we automatically assign `joint_i` names if joints are not passed? Or would that break stuff?

Reply:
I think that would break stuff.  There are definitely checks here and there on whether a configuration has names attached.  And if the user hasn't updated those names to the real things...

for example in `robot.merge_group_with_full_configuration`....

Reply:
Ok, then let's not do that ;)
But I still find it unexpected/exotic that `bool` is defined in terms of values but `len` is defined in terms of names. As a user, you have no way of guessing this difference unless you read the code. Would it be an option to define `len` in terms of values instead and maybe raise an exception if `len(joint_values) != len(joint_names)`?

Reply:
I feel pretty strongly that `len` should reflect the number of elements encountered while iterating over the object, in this case, the `joint_names`.  and i don't like the idea of `__len__` raising an exception.  but i am at a loss as to why i decided `bool` should be determined by `joint_values`.  there is no test (that i didn't write) that fails for this.  maybe `__bool__` should be left undefined.

Reply:
agree that it makes more sense than `__len__` is based on `joint_names` because it's part of the dict interface. Undefining `__bool__` would be an option if you don't rely on that behavior for anything else.

Reply:
oh, `if not config` is used in the vrep client `set_robot_config`, so leaving `__bool__` undefined fails there.  should it always return `True`?

Reply:
but maybe those usages could be replaced for `if config is None`... at least a quick look at both `set_robot_config` and `set_robot_pose` seem to indicate both are intended to be null-checks.

Reply:
oh, right, of course, if `__bool__` is not defined, then the default `__bool__` calls `__len__`, so that's why i did it the way i did... because if `joint_names` is empty, that doesn't mean the `config` should be falsy.

Reply:
So I think it makes the most sense to have `__bool__` always return `True` which is the default behavior when there is no overwriting of `__bool__` and `__len__`.

Reply:
and then i don't have to worry about breaking things in the v-rep client or elsewhere.

Reply:
👍 I would definitely add tests and an inline comment about why these two return what they do (as summary of this thread basically), otherwise it's a given that we'll forget and break it next time we touch it.